In [115]:
import numpy as np
import pandas as pd

### Задача 1. 
Даны значения зарплат из выборки выпускников: 100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150. Посчитать (желательно без использования статистических методов наподобие std, var, mean) среднее арифметическое, среднее квадратичное отклонение, смещенную и несмещенную оценки дисперсий для данной выборки.

In [12]:
df = np.array([100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150])

In [14]:
df.sort()

In [15]:
df

array([ 17,  24,  25,  30,  33,  45,  55,  57,  65,  65,  70,  75,  75,
        77,  80,  84,  89,  90, 100, 150])

Среднее арифметическое равно:

In [31]:
list = [ 17,  24,  25,  30,  33,  45,  55,  57,  65,  65,  70,  75,  75,
        77,  80,  84,  89,  90, 100, 150]

sum = 0
n = 20

for num in list:
    sum += num

In [64]:
average = sum/n
print(average)

65.3


Среднее квадратичное отклонение:

In [74]:
from math import sqrt
list = [ 17,  24,  25,  30,  33,  45,  55,  57,  65,  65,  70,  75,  75,
        77,  80,  84,  89,  90, 100, 150]

x1 = 0

for item in list:
    x1 = x1 + (item - average)**2
    sko = sqrt(x1/len(list))

In [81]:
print(sko)

30.823854398825596


Смещенная и несмещенная оценки дисперсий для данной выборки:

In [88]:
list = [ 17,  24,  25,  30,  33,  45,  55,  57,  65,  65,  70,  75,  75,
        77,  80,  84,  89,  90, 100, 150]

x1 = 0

for item in list:
    x1 = x1 + (item - average)**2
    S1 = x1/len(list)

In [89]:
S1

950.11

In [90]:
list = [ 17,  24,  25,  30,  33,  45,  55,  57,  65,  65,  70,  75,  75,
        77,  80,  84,  89,  90, 100, 150]

x2 = 0

for item in list:
    x2 = x1 + (item - average)**2
    S2 = x1/(len(list) - 1)

In [91]:
S2

1000.1157894736842

### Задача 2. 
В первом ящике находится 8 мячей, из которых 5 - белые. Во втором ящике - 12 мячей, из которых 5 белых. Из первого ящика вытаскивают случайным образом два мяча, из второго - 4 мяча. Какова вероятность того, что 3 мяча белые?

Формула полной вероятности - P = P(A/B1) * P(B1) + P(A/B2) * P(B2) + P(A/B3) * P(B3)
P(B1) - вероятность что не достанем ни одного белого мяча из корзины 1, P(A/B1) - вероятность того что достанем 3 белых мяча из корзины два при выполнении предыдущего условия, P(B2) - вероятность что достанем один белый мяч из корзины 1, P(A/B2) - вероятность того что достанем 2 белых мяча из корзины два при выполнении предыдущего условия, P(B3) - вероятность что достанем два белых мяча из корзины 1, P(A/B3) - вероятность того что достанем 1 белый мяч из корзины два при выполнении предыдущего условия.

In [116]:
from math import factorial

In [120]:
def combinations(k, n1):
    return factorial(n1)/(factorial(k) * factorial(n1 - k))

In [128]:
C34 = combinations(3, 4)
C312 = combinations(3, 12)
C34/C312

0.01818181818181818

In [129]:
C05 = combinations(0, 5)
C28 = combinations(2, 8)
C05/C28

0.03571428571428571

In [130]:
C24 = combinations(2, 4)
C312 = combinations(3, 12)
C24/C312

0.02727272727272727

In [131]:
C15 = combinations(1, 5)
C28 = combinations(2, 8)
C15/C28

0.17857142857142858

In [134]:
C14 = combinations(1, 4)
C312 = combinations(3, 12)
C14/C312

0.01818181818181818

In [135]:
C25 = combinations(2, 5)
C28 = combinations(2, 8)
C25/C28

0.35714285714285715

In [137]:
P = (C34/C312) * (C05/C28) + (C24/C312) * (C15/C28) + (C14/C312) * (C25/C28)
P

0.012012987012987014

Вероятность того, что 3 мяча белые примерно равна 1.2 %

### Задача 3. 
На соревновании по биатлону один из трех спортсменов стреляет и попадает в мишень. Вероятность попадания для первого спортсмена равна 0.9, для второго — 0.8, для третьего — 0.6. Найти вероятность того, что выстрел произведен: a). первым спортсменом б). вторым спортсменом в). третьим спортсменом.

Будем использовать формулу Байеса, так как необходимо найти вероятность того какой спортсмен произвел выстрел, при том что сам выстрел уже произведен(то есть найти вероятность события B, при том что событие A уже произошло). 

Формула Байеса: P(B1/A) = (P(A/B1) * P(B1))/P(A), где P(B1/A) - вероятность того, что произведенный выстрел сделал n-й спортсмен, P(A/B1) - вероятность того что будет произведен выстрел при условии что зафиксирован стрелок, P(B1) - вероятность попадания спортсмена, P(A) - полная вероятность что выстрел произведен.

P(A) = P(A/B1) * P(B1) + P(A/B2) * P(B2) + P(A/B3) * P(B3)

In [152]:
def combinations(x1, y1, x2, y2, x3, y3):
    return x1 * y1 + x2*y2 + x3*y3

In [155]:
PA = combinations(1/3, 0.9, 1/3, 0.8, 1/3, 0.6)
PA

0.7666666666666666

P(B1/A) = (P(A/B1) * P(B1))/P(A)

In [157]:
def combinations(x, y, z):
    return (x*y)/z

In [159]:
P1 = combinations(1/3, 0.9, 0.77)
P1

0.3896103896103896

In [160]:
P2 = combinations(1/3, 0.8, 0.77)
P2

0.3463203463203463

In [161]:
P3 = combinations(1/3, 0.6, 0.77)
P3

0.2597402597402597

Вероятности примерно равны 39 %, 35% и 26%

### Задача 4. 
В университет на факультеты A и B поступило равное количество студентов, а на факультет C студентов поступило столько же, сколько на A и B вместе. Вероятность того, что студент факультета A сдаст первую сессию, равна 0.8. Для студента факультета B эта вероятность равна 0.7, а для студента факультета C - 0.9. Студент сдал первую сессию. Какова вероятность, что он учится: a). на факультете A б). на факультете B в). на факультете C?

Как и в предыдущей задаче воспользуемся формулой Байеса. 

Формула Байеса: P(B1/A) = (P(A/B1) * P(B1))/P(A), где P(B1/A) - вероятность того, что студент сдавший экзамен учится на n-м факультете, P(A/B1) - вероятность того что студент поступил при условии что зафиксирован факультет, P(B1) - вероятность студента сдать сессию, P(A) - полная вероятность что экзамен сдан.

P(A) = P(A/B1) * P(B1) + P(A/B2) * P(B2) + P(A/B3) * P(B3)

In [164]:
def combinations(x1, y1, x2, y2, x3, y3):
    return x1 * y1 + x2*y2 + x3*y3

x1, x2, x3 - будут равны 0.25, 0.25 и 0.5 соответсвенно. 

In [165]:
PA = combinations(0.25, 0.8, 0.25, 0.7, 0.5, 0.9)
PA

0.825

P(B1/A) = (P(A/B1) * P(B1))/P(A)

In [166]:
def combinations(x, y, z):
    return (x*y)/z

In [167]:
Pa = combinations(0.25, 0.8, 0.825)
Pa

0.24242424242424246

In [168]:
Pb = combinations(0.25, 0.7, 0.825)
Pb

0.21212121212121213

In [169]:
Pc = combinations(0.5, 0.9, 0.825)
Pc

0.5454545454545455

Вероятности примерно равны 24.2 %, 21.2% и 54.6%

### Задача 5. 
Устройство состоит из трех деталей. Для первой детали вероятность выйти из строя в первый месяц равна 0.1, для второй - 0.2, для третьей - 0.25. Какова вероятность того, что в первый месяц выйдут из строя: а). все детали б). только две детали в). хотя бы одна деталь г). от одной до двух деталей?

1) Выйдут из строя все детали

In [195]:
P1 = 0.1
P2 = 0.2
P3 = 0.25
P = P1 * P2 * P3

In [196]:
result = f"{P:.1%}"
result

'0.5%'

2. Выйдут из строя только две детали

In [204]:
P = P1 * P2 * (1 - P3) + P1 * P3 * (1 - P2) + P2 * P3 * (1 - P1)

In [205]:
result = f"{P:.1%}"
result

'8.0%'

3. Выйдет из строя хотя бы одна деталь

In [206]:
P = 1 - (1 - P1) * (1 - P2) * (1 - P3)
result = f"{P:.1%}"
result

'46.0%'

4. Выйдет из строя от одной до двух деталей

In [215]:
P = P1 * (1 - P2) * (1 - P3) + P2 * (1 - P1) * (1 - P3) + P3 * (1 - P1) * (1 - P2) +\
    P1 * P2 * (1 - P3) + P1 * P3 * (1 - P2) + P2 * P3 * (1 - P1)

In [219]:
result = f"{P:.1%}"
result

'45.5%'